enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,363 @@
+
+The OVN-K master will also watch the ``DNSName`` CRs. Whenever the ``.status`` of the CRs will be updated with new IPs and corresponding
+TTL information for a DNS name, OVN-K master will update the ``AdressSet`` mapped to the DNS name. This ``AdressSet`` will be linked
+to the ACL rule(s) for the EgressFirewall rule(s) in which the DNS name is used. This will ensure that the latest IPs are always updated in
+the ``AddressSets``.
+
+The OVN-K master will not directly query for the wildcard DNS names, rather it will query for the DNS names that get added to the
+``.status`` of the corresponding ``DNSName`` CR. However, the list of the DNS names to lookup for a wildcard DNS name should also
+not become stale if a DNS name belonging its subdomain is removed. To achieve this a retry counter will be used for the DNS name
+lookups. If the lookup fails for a DNS name listed in the ``.status`` of a ``DNSName`` CR for threshold number of times (say 5),
+then the DNS name will be removed from the ``.status``.
+
+
+### Risks and Mitigations
+
+
+### Drawbacks
+
+* Whenever there's a change in the IPs or the next lookup time for a DNS name, the additional step of updating the
+related ``DNSName`` CRs will be executed. This might add some delay to the DNS lookup process. However, this will only
+happen whenever there's a change in the DNS information.
+
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+* This enhancement will be tested through e2e tests by adding EgressFirewall rules containing regular DNS names
+and wildcard DNS names.
+* Testing the feature where IPs are dynamically changed may be a little bit tricky as this will probably include
+creation of DNS records and then changing the IP adresses of the DNS records through the e2e tests.
+
+### Graduation Criteria
+
+This is a user facing change and will directly go to GA. This feature requires an update to Openshift Docs.
+
+#### Dev Preview -> Tech Preview
+
+N.A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N.A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+* On upgrade, the OVN-K master will create the corresponding ``DNSName`` CRs for each DNS name in the
+existing EgressFirewall resources. The ``egressfirewall`` plugin will also start updating the ``.status``
+fields of the ``DNSName`` CRs. The scenarios arising out of the order of the update of the various components
+are dicussed in [Version Skew Strategy](#version-skew-strategy)
+
+Downgrade expectations:
+* On downgrade, the ``DNSName`` CRs may still remain. However, these CRs would not have any impact on how
+EgressFirewall ACL rules are implemented in the downgraded cluster. Deleting the CR Definition of ``DNSName``
+from the cluster would remove all the ``DNSName`` CRs.
+
+### Version Skew Strategy
```

Comment:
one component has to create the CRD, and the other component has to deal with the possibility that the CRD doesn't exist when it starts up, but is created later

Reply:
Will the components create the CRD or the CVO?